assessments/projects/knative/self-assessment.md

```diff
@@ -0,0 +1,577 @@
+
+Event sources are created by instantiating a CR from a Source object. The Source object defines the arguments and parameters needed to instantiate a CR.
+
+(maintainer suggestions: Event sources look like the Broker controller + egress; an event sink would look like a controller + ingress.  (I.e. an event source looks like a Broker or Channel which is "magically" receiving messages from somewhere else and then sending them into Knative Eventing.))
+
+**Knative Functions** (https://knative.dev/docs/functions/):
+
+Knative Functions is an extension of the Knative ecosystem, designed to simplify the deployment and management of serverless functions. Similar to Knative Eventing, it has several key components and features that enable efficient and scalable function execution in a cloud-native environment:
+
+
+1. Serverless Function Deployment: Knative Functions focuses on the deployment of serverless functions, allowing developers to write code in their language of choice. These functions are automatically scaled up or down based on demand, ensuring efficient resource utilization.
+
+
+2. Integration with Knative Eventing: Functions in Knative can be triggered by events managed by Knative Eventing. This integration enables a seamless flow where functions are executed in response to various events, enhancing the event-driven architecture.
+
+
+3. Simplified Function Management: The platform provides tools and APIs that simplify the process of deploying, updating, and managing functions. This includes straightforward CLI tools and declarative configurations, making it easier for developers to handle their serverless functions.
+
+
+4. Support for Multiple Languages: Knative Functions supports a wide range of programming languages, allowing developers to write functions in the language they are most comfortable with. This flexibility facilitates the adoption of serverless architecture across different development teams.
+
+
+5. Built-In Observability: Knative Functions come with built-in monitoring and logging capabilities. This allows developers to easily track the performance of their functions and diagnose issues, ensuring the reliability and stability of the functions.
+
+
+6. Event Source Integration: Functions can be directly integrated with various event sources, allowing them to react to a wide range of events from different sources. This makes Knative Functions ideal for building complex, event-driven applications.
+
+
+7. Container-Based Function Execution: Functions in Knative are executed in containers, leveraging Kubernetes' container orchestration capabilities. This approach ensures portability, scalability, and consistent execution environments for functions.
+Further detailed Eventing components
+
+### Actions
```

Comment:
The intent of this section is to showcase what each of the project performs in order to provide some service or functionality, however what I see here is what functionalities knative has and how to use knative. Please update this section to suit actions per our template - https://github.com/cncf/tag-security/blob/main/assessments/guide/self-assessment.md#actions.

A good reference is https://github.com/cncf/tag-security/blob/main/assessments/projects/pixie/self-assessment.md#actions